File: src/main/java/work/xeltica/craft/core/modules/xphone/XphoneCommand.java
package work.xeltica.craft.core.modules.xphone;

import org.bukkit.Sound;
import org.bukkit.SoundCategory;
import org.bukkit.command.Command;
import org.bukkit.command.CommandSender;
import org.bukkit.entity.Player;

import net.kyori.adventure.text.Component;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import work.xeltica.craft.core.api.commands.CommandPlayerOnlyBase;
import work.xeltica.craft.core.modules.hint.Hint;
import work.xeltica.craft.core.modules.hint.HintModule;
import work.xeltica.craft.core.modules.item.ItemModule;

import java.util.List;
import java.util.Objects;

/**
 * X Phoneを受け取るコマンド
 * @author Xeltica
 */
public class XphoneCommand extends CommandPlayerOnlyBase {
    @Override
    public boolean execute(Player player, Command command, String label, String[] args) {
        final var item = ItemModule.INSTANCE.getItem(ItemModule.ITEM_NAME_XPHONE);
        player.getInventory().addItem(item);
        player.playSound(player.getLocation(), Sound.ENTITY_ITEM_PICKUP, SoundCategory.PLAYERS, 1, 1);
        player.sendMessage(Objects.requireNonNull(item.getItemMeta().displayName()).append(Component.text("を付与しました")));
        HintModule.INSTANCE.achieve(player, Hint.TWIN_XPHONE);
        return true;
    }

    @Nullable
    @Override
    public List<String> onTabComplete(@NotNull CommandSender commandSender, @NotNull Command command, @NotNull String label, String[] args) {
        return COMPLETE_LIST_EMPTY;
    }
}

File: src/main/java/work/xeltica/craft/core/modules/xphone/XphoneCommand.kt
package work.xeltica.craft.core.modules.xphone

import net.kyori.adventure.text.Component
import org.bukkit.Sound
import org.bukkit.SoundCategory
import org.bukkit.command.Command
import org.bukkit.command.CommandSender
import org.bukkit.entity.Player
import work.xeltica.craft.core.api.commands.CommandPlayerOnlyBase
import work.xeltica.craft.core.modules.item.ItemModule.getItem

/**
 * X Phoneを受け取るコマンド
 * @author Lutica
 */
class XphoneCommand : CommandPlayerOnlyBase() {
    private val subCommands = listOf("get")
    override fun execute(player: Player, command: Command, label: String, args: Array<out String>): Boolean {
        if (args.isNotEmpty() && args[0] == "get") {
            // getが指定されたら、アイテムを受け取る
            val item = getItem(XphoneModule.ITEM_NAME_XPHONE)
            player.inventory.addItem(item)
            player.playSound(player.location, Sound.ENTITY_ITEM_PICKUP, SoundCategory.PLAYERS, 1f, 1f)
            val displayName = item.itemMeta.displayName() ?: item.displayName()
            player.sendMessage(displayName.append(Component.text("を付与しました")))
        } else {
            XphoneModule.openSpringBoard(player)
        }
        return true
    }

    override fun onTabComplete(commandSender: CommandSender, command: Command, label: String, args: Array<String>): List<String> {
        return if (args.size > 1) COMPLETE_LIST_EMPTY else subCommands
    }
}